kornia/core/image.py

```diff
@@ -0,0 +1,280 @@
+
+    .. note::
+
+        Disclaimer: This class provides the minimum functionality for image manipulation. However, as soon
+        as you start to experiment with advanced tensor manipulation, you might expect fancy
+        polymorphic behaviours.
+
+    .. warning::
+
+        This API is experimental and might suffer changes in the future.
+
+    Args:
+        data (torch.Tensor): a torch.Tensor with the shape of :math:`(B)(C, H, W)`.
+        color (ImageColor): the color space representation of the input data.
+        mean (List[float]): a list containing the mean values of the input data.
+        std (List[float]): a list containing the standard deviation of the input data.
+
+    Examples:
+        >>> import kornia
+        >>> import torch
+        >>> import numpy as np
+        >>>
+        >>> # from a torch.tensor
+        >>> data = torch.randint(0, 255, (3, 4, 5))  # CxHxW
+        >>> img = Image(data, ImageColor.RGB)
+        >>> assert img.channels == 3
+        >>> assert img.resolution == (4, 5)
+        >>> assert (img.data == data).all()
+        >>> out = (img + 2 * img - 3 * img.clone()).sum()
+        >>> assert isinstance(out, Image)
+
+        >>> # from a bached torch.tensor
+        >>> data = torch.rand(2, 3, 4, 5)  # BxCxHxW
+        >>> img = Image(data, ImageColor.RGB)
+        >>> assert img.channels == 3
+        >>> assert img.resolution == (4, 5)
+        >>> assert img.is_batch
+        >>> r, g, b = img[:, 0], img[:, 1], img[:, 2]
+        >>> mean_ch = img.mean(dim=1, keepdim=True)
+        >>> mean_ch.shape
+        torch.Size([2, 1, 4, 5])
+
+        >>> # from a numpy array (like opencv)
+        >>> data = np.ones((4, 5, 3), dtype=np.uint8)  # HxWxC
+        >>> img = Image.from_numpy(data, color=ImageColor.BGR)
+        >>> assert img.channels == 3
+        >>> assert img.resolution == (4, 5)
+
+        >>> # direct conversion from/to dlpack
+        >>> data = torch.rand(3, 4, 5)
+        >>> dlpack = Image(data, ImageColor.RGB).to_dlpack()
+        >>> img = Image.from_dlpack(dlpack, ImageColor.RGB)
+
+        >>> # apply any kornia function
+        >>> data = torch.rand(3, 4, 5)
+        >>> img = Image(data, color=ImageColor.RGB)
+        >>> img_gray = img.apply(kornia.color.rgb_to_grayscale)
+        >>> assert isinstance(img_gray, Image)
+        >>> assert img_gray.color == ImageColor.GRAY
+
+        >>> # denormalize for visualization
+        >>> data = torch.rand(3, 4, 5)
+        >>> img = Image(data, color=ImageColor.RGB, mean=[0.,0.,0], std=[255.,255.,255.])
+        >>> img_denorm = img.denormalize()
+        >>> assert img_denorm.mean().item() > 1.0
+    """
+    # defaults
+    _meta: Dict[str, Any] = {}
+    _meta['color'] = ImageColor.RGB
+    _meta['mean'] = None
+    _meta['std'] = None
+
+    @staticmethod
+    def __new__(
+        cls,
+        data: Tensor,
+        color: ImageColor,
+        mean: Optional[List[float]] = None,
```

Comment:
tensor or list?

Reply:
it could be both -- but as discussed with @lferraz I guess the most common use case is to have a single mean/std per dataset (usually one per script too). I want to keep minimal an iterate over use cases. If you have one specific, then totally fine.